lib/rest-adapter.js

```diff
@@ -105,6 +104,27 @@ RestAdapter.prototype.connect = function(url) {
   this.connection = url;
 };
 
+/**
+ * Get the auth from the remotes or from the
+ * loopback options if available
+ * @param remotes
+ * @param args
+ * @private
+ */
+RestAdapter.prototype._extractAuth = function(remotes, args) {
```

Comment:
On the second thought, this implementation is very loopback-specific. We are trying to keep strong-remoting decoupled from loopback itself - this package should provide extension hooks allowing loopback to inject loopback-specific behaviour.

In this particular case, the function parsing the accessToken from the options argument should live in LoopBack. It can be provided to RestAdapter for example via the `options` object which contains other related configuration too.

Reply:
The main difference for this one, is that it is executed per request. E.g. for access tokens it changes for each different user hitting the server, that is why it doesn't fit in the other options object that is used to initialise the adapter. 

I'll think about it to see if I can set this through loopback code. One option is making it like a basic function to get the auth, that I can override in some sort of boot script that iterates over all datasources and sets the more complex extractAuth programmatically on the adapter. Apart from this boot script solution feeling a bit hacky, I was also hoping that this might have been useful to developers that want loopback servers to hit other loopback servers securely (e.g. microservices). I was hoping it would have worked out of the box (after setting the config flag). Open to ideas here.

Reply:
I did your solution below of using the meta to find the options. All it is looking for is an accessToken in an options object. I made a generic way to get an arg by name on RestMethod.